src/Compilers/CSharp/Portable/Symbols/NamespaceSymbol.cs

```diff
@@ -311,6 +311,23 @@ internal NamespaceSymbol GetNestedNamespace(NameSyntax name)
             return null;
         }
 
+        private ImmutableArray<NamedTypeSymbol> _lazyTypesWithExtensionMethods;
+        private ImmutableArray<NamedTypeSymbol> TypesWithExtensionMethods
+        {
+            get
+            {
+                var typesWithExtensionMethods = this._lazyTypesWithExtensionMethods;
+                if (typesWithExtensionMethods.IsDefault)
+                {
+                    this._lazyTypesWithExtensionMethods = this.GetTypeMembersUnordered().WhereAsArray(t => t.MightContainExtensionMethods);
```

Comment:
@gafter The JIT inserts a memory barrier after each write to a field of reference type. So technically we don't need another one. That said I believe we decided a while ago to use interlocked across the code base to make it explicit. It would be interesting to see if all the interlocked operations actually slow the compiler down significantly.